### DISCUSSION TITLE
Metrics for subscriptions?

### DISCUSSION THREAD

**Discussion by vkuznecovas** [emqx/emqx#12609] [Q&A] [answered]
I've seen that the EMQX REST API exposes metrics per topic, but this does not seem to work with wildcard topics.

I'd like to be able to see the combined metrics for such topics. In this case, I have 50k+ topics with such wildcards. I've tried to query them via the API using the `List topic metrics` endpoint, but that flat out refuses to respond in any meaningful amount of time, so I gave up on waiting. 

I'd like to have a metric that I can use to automatically scale the subscriber pods in a k8s install, but can't seem to find one that would work. The global metrics are not suited for this, as I have other topics that influence these metrics and would be misleading. 

Are there any metrics for shared subscriptions? Perhaps there is another way to achieve this?

**Comment by zmstone** [comment, marked as answer]
Hi @vkuznecovas 
The current implementation is indeed limited to PUBLISH topics, but not for topic-filters.
To support wildcards, it's technically not complicated, but I am not sure when we can get this prioritized.

You can go ahead to the issue tab and create a "feature" request.

**Comment by zmstone** [comment]
> Are there any metrics for shared subscriptions? Perhaps there is another way to achieve this?

Sorry, not that I can think of.

**Comment by vkuznecovas** [comment]
Thanks, FR opened: https://github.com/emqx/emqx/issues/12614
